src/lib/kw_db.sh

```diff
@@ -232,6 +229,75 @@ function select_from()
   cmd_manager "$flag" "$cmd"
 }
 
+# This function gets the values in the table of given database
+# with the given conditions
+#
+# @table:     Table to select info from
+# @columns:   Columns of the table to get
+# @pre_cmd:   Pre command to execute
+# @_condition_array: An array reference of condition pairs
+# @flag:      Flag to control function output
+# @db:        Name of the database file
+# @db_folder: Path to the folder that contains @db
+#
+# Return:
+# 2 if db doesn't exist; 22 if table is empty
+# 0 if succesful; non-zero otherwise
+function select_from_where()
+{
+  local table="$1"
+  local columns="${2:-"*"}"
+  local pre_cmd="$3"
+  local -n _condition_array="$4"
+  local flag=${5:-'SILENT'}
+  local db="${6:-"$DB_NAME"}"
+  local db_folder="${7:-"$KW_DATA_DIR"}"
+  local where_clause=''
+  local db_path
+  local query
+
+  db_path="$(join_path "$db_folder" "$db")"
+
+  if [[ ! -f "$db_path" ]]; then
+    complain 'Database does not exist'
+    return 2
+  fi
+
+  if [[ -z "$table" || -z "${!_condition_array[*]}" ]]; then
+    complain 'Empty table or condition array.'
+    return 22 # EINVAL
+  fi
+
+  where_clause="$(generate_where_clause '_condition_array')"
+  query="SELECT ${columns} FROM ${table} WHERE ${where_clause} ;"
+
+  cmd="sqlite3 -init "${KW_DB_DIR}/pre_cmd.sql" \"${db_path}\" -batch \"$query\""
+  cmd_manager "$flag" "$cmd"
+}
+
+function generate_where_clause()
+{
+  local -n c_array="$1"
```

Comment:
I assume you are using this name because the use of `condition_array` or `_condition_array` probably causes problems (Bash, sometimes you are complicated :cry:). However, find a more descriptive name, like `conditions_array_ref` (surely there is something better than this :rofl:).